### PR TITLE
Edit mode 885

### LINKED ISSUE #885: UI Edit mode for everything, not only for top tags
Edit mode works for top tags very well, and I think it could works for product page as well. Also It will give ability for administrators to see the site as regular guest do. 

@aaronjudd @mikemurray, what do you think?

### PR DESCRIPTION
**Pull request by mikemurray**
Resolves #885 

Remove global edit mode button in favor of local, in-context edit buttons.

Updated React Button component Added IconButton, and EditButton components

Removed hardcoded edit buttons in various places in favor of the component `EditButton`.

Added active state to all edit buttons.

### CODE REVIEW

**Comment by aaronjudd** [comment]
```debug.js:41 Exception in template helper: TypeError: Cannot read property 'template' of undefined
    at Object.SocialEditButton (http://localhost:3000/app/app.js?hash=40c69877e6da5b8404989e3a53db27e3c6ced2e6:2913:43)
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:2994:16
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:1653:16
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3046:66
    at Function.Template._withTemplateInstanceFunc (http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3687:12)
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3045:27
    at Object.Spacebars.call (http://localhost:3000/packages/spacebars.js?hash=65db8b6a8e3fca189b416de702967b1cb83d57d5:172:18)
    at http://localhost:3000/app/app.js?hash=40c69877e6da5b8404989e3a53db27e3c6ced2e6:2028:26
    at wrappedArgFunc (http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:2903:14)
    at .<anonymous> (http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:2616:26)
debug.js:41 Exception in template helper: TypeError: Cannot read property 'data' of undefined
    at variantIsInActionView (http://localhost:3000/app/app.js?hash=40c69877e6da5b8404989e3a53db27e3c6ced2e6:1527:51)
    at Object.EditButton (http://localhost:3000/app/app.js?hash=40c69877e6da5b8404989e3a53db27e3c6ced2e6:1592:19)
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:2994:16
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:1653:16
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3046:66
    at Function.Template._withTemplateInstanceFunc (http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3687:12)
    at http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:3045:27
    at Object.Spacebars.call (http://localhost:3000/packages/spacebars.js?hash=65db8b6a8e3fca189b416de702967b1cb83d57d5:172:18)
    at http://localhost:3000/app/app.js?hash=40c69877e6da5b8404989e3a53db27e3c6ced2e6:1487:24
    at wrappedArgFunc (http://localhost:3000/packages/blaze.js?hash=ef41aed769a8945fc99ac4954e8c9ec157a88cea:2903:14)```

**Comment by mikemurray** [comment]
This should be fixed now.

File: imports/plugins/core/ui/client/components/button/editButton.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @param {Object} props Props passed into component
+ * @returns {IconButton} Retruns an IconButton component with pre-configured icons for editing
+ */
+const EditButton = (props) => {
+  return (
+    <IconButton
+      icon="fa fa-pencil"
+      onIcon="fa fa-check"
+      toggle={true}
+      {...props}
```
Comment (kieckhafer) [inline imports/plugins/core/ui/client/components/button/editButton.js]: Is this OK in a non "jsx" file?

**Comment by kieckhafer** [comment]
Overall looks good to me, there are a few `...props` and `...attrs` in non "jsx" files, is that OK?

**Comment by aaronjudd** [comment]
👍  absolutely fabaloose

[Approved with PullApprove](https://pullapprove.com/reactioncommerce/reaction/pull-request/1217/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=reactioncommerce/reaction)

**Comment by kieckhafer** [comment]
👍

**Pull request merged**